test/main_dialog_spec.rb

```diff
@@ -112,12 +113,17 @@ def expect_update_table_with(expected_items)
       dialog.run
     end
 
-    it "updates the alternatives table" do
+    it "updates the modified alternative on the table" do
       allow(UpdateAlternatives::AlternativeDialog).to receive(:new)
         .and_return(alternative_dialog)
       allow(alternative_dialog).to receive(:run)
 
-      expect(Yast::UI).to receive(:ChangeWidget).with(:alternatives_table, :Items, kind_of(Array))
+      allow(Yast::Term).to receive(:new).and_call_original
+      allow(Yast::Term).to receive(:new).with(:id, :alternatives_table).and_return(table_id)
```

Comment:
why those two calls are needed? term construction should be safe and should not need mock.